cr-db/tests/test_db.py

```diff
@@ -52,5 +100,32 @@ def test_select_with_filter():
        results.  Be careful about the "missing" (None) data.
 
        Answer but don't code: what would a generic solution look like to compare any columns containing categories?
+       You can use the ANOVA stats method or the t.test method from the scipy library, passing the pairwise column you
+       wanna elaborate for statistical inference
+    """
+
+    df = pd.read_csv('data/S-O-1k.csv', header=0)
+
+    female_df = df[df['Combined Gender'] == 'Female']
+    female_df = female_df.loc[:, ['FormalEducation', 'SalaryAdjusted']]
+
+    rt = np.dtype([('formal_education', np.str_, 140), ('salary_adj', np.float32)])
+    s_array = np.array(list(zip(*[female_df[c].values.tolist() for c in female_df])), dtype=rt)
+    map_v = {c: v for v, c in enumerate(np.unique(s_array['formal_education']))}
+    sv_array = s_array[~np.isnan(s_array['salary_adj'])]
+
+    final_array_type = ([('formal_education', np.str_, 200),
+                         ('mean', np.float64),
+                         ('var', np.float64),
+                         ('std', np.float64)])
+
+    final = []
+    for k in map_v:
+        final.append((k, np.mean(sv_array['salary_adj'][sv_array['formal_education'] == k]),
+                      np.var(sv_array['salary_adj'][sv_array['formal_education'] == k]),
+                      np.std(sv_array['salary_adj'][sv_array['formal_education'] == k])))
+
+    final = np.array(final, dtype=final_array_type)
+    print(final)
```

Comment:
> does this seem correct?
> 
> ("Some college/university study without earning a bachelor's degree", nan, nan, nan)
> 
> I can't remember if our data is all missings for those or not...
I checked in the 1k file and there's no salary information (for Women) for 1 class of Formal Education.
![image](https://user-images.githubusercontent.com/4196091/54837542-f342fd00-4cc6-11e9-9844-7ffa5f49a375.png)



> instead of a print and an assert 0, could you codify the expected results? Why or why not?

i'm working on that right now

Reply:
solved in e7dc6ee84950b57c5d0ee39fce9d599a0694f19a